### ISSUE TITLE
Rename org/repositories to achieve consistency

### ISSUE THREAD

**Issue by a-teammate** [inexorgame/inexor-core#404]
Since we introduced the modules "InexorCore" and "InexorFlex" for the C++ and nodejs parts respectively, I think our InexorCore-repo ("inexor-game/code") should be renamed to "inexor-game/core" to be consistent.

**Comment by aschaeffer** [comment]
Not Found.

**Comment by Croydon** [comment]
This breaks very old links, since it would be our second rename (we renamed our organisation once), the question is, if we do care for them or if it doesn't matter.

As @Terencode suggested once, many orgs have their main repo named like their product itself, e.g. https://github.com/atom/atom
https://github.com/desktop/desktop
https://github.com/eclipse/eclipse
https://github.com/flatpak/flatpak

So I would be rather in favor of "inexor" as the name for the core repository instead of actually core.

Bringing the discussion about the **organisation name** from Riot here:
We couldn't get the name "inexor" because it's taken here on GitHub AND on way too many platforms already. However, we are using "inexorgame" everywhere except on GitHub (Travis and AppVeyor are using the GitHub names, so that is a consequence rather than a choice) and on GitHub "inexor-game".

If we should decide to rename the core **repository**, and therefore breaking already the first links, we could do now a second cut and rename the **organisation** to "inexorgame". The pain of breaking links will only get bigger with time. So pretty much now or never.

AFAIK GitHub is only upholding a single redirect from the latest renaming (as long as nobody else wants to use the name).  So if we do both we should first rename the repo, then the organisation, to get better found by people who are getting a 404 via URLs -> if the repo is not found people are likely trying to see if the organisation is still existing, and they should be getting redirect to the new name.

**Comment by Croydon** [comment]
It is really good sarcasm that this issue got the number #404.

**Comment by a-teammate** [comment]
The sauerfork redirect is dead anyways. Or am I doing sth wrong?
"sauerbraten-fork" "sauerfork" and "sauerbratenfork" are all not working

**Comment by aschaeffer** [comment]
We could name it:
https://github.com/inexor-game/inexor-core (or better: https://github.com/inexorgame/inexor-core)
https://github.com/inexor-game/inexor-flex (or better: https://github.com/inexorgame/inexor-flex)

If we are going to rename the organization and links are broken, we can also rename the repos.

After cloning, you would have a folder named `inexor-core` and `inexor-flex` instead of `code` and `flex` which is more meaningful and avoids conflicts.

**Comment by Croydon** [comment]
Okay, I just tested it and it seems like the redirects are more powerful as I thought.
https://github.com/sauerbraten-fork/sauerbraten-fork/ is still redirecting here.

https://github.com/sauerbraten-fork/ is not, but most of the time we link to our main repository anyway I guess.

So our current URL of our main repository should redirect to whatever we rename our org+repo.

>We could name it:
https://github.com/inexor-game/inexor-core

I don't like the "inexor-" prefix, why adding it to our main repo and flex but not e.g. website, data and so on? Seems redudant, however, naming the main repository simply "inexor" alone seems to be very common. I would leave flex as it is.

**Comment by a-teammate** [comment]
"inexorflex" is shit. Imagine you don't know the name "inexor" yet. How would you react on this?  
Imo the same for "inexorgame", it's clear for us but not as clear as "inexor-game" to read. But anyways..


Plz not "inexorflex". at least a dash should lead the eyes of the novice reader a bit.

@Croydon

Bc website and data are not our core products while flex and core are both (besides of the issue tracker and the wiki in the repo) more or less equally

Good to know the repo redirect works :)

**Comment by aschaeffer** [comment]
To avoid any confusions it would be fine if the repository name of `Inexor Core` has something with `core` in it's name. Just `inexor` isn't precise enough. And `core` could conflict with repositories named `core` from other projects (search for https://github.com/search?utf8=%E2%9C%93&q=core&type=)

**Comment by aschaeffer** [comment]
Ok, the dash makes sense. Therefore I would prefer https://github.com/inexor-game/inexor-core and https://github.com/inexor-game/inexor-flex

**Comment by Croydon** [comment]
> Bc website and data are not our core products while flex and core are both  

Maybe for the website, but data is absolutely needed, in my opinion it would be another inconsistency if two, but only two, repositories do have the "inexor-" prefix. It is also more to type and remember.  

> And core could conflict with repositories named core from other projects (search for https://github.com/search?utf8=%E2%9C%93&q=core&type=)  

What kind of conflict? Every half-mature git client should check that it doesn't overwrite another repository and if, optionally, the developer doesn't like the name, he can simply rename it.  

> Ok, the dash makes sense. Therefore I would prefer https://github.com/inexor-game/inexor-core and https://github.com/inexor-game/inexor-flex  

If you meet someone and want to tell them go to https:/.... it is easier to remember when there is no dash in there and no second "inexor". Keep it short and simple.  

Therefore I'm definitely for it to remove the dash **at least** in the organisation name. Making it consistent with other inexor accounts.

**Comment by a-teammate** [comment]
KISS would be: go 2 inexor.org

**Comment by aschaeffer** [comment]
For me it's OK to remove the dash in the organization name if we can harmonize our account names. Seems to be easier than to rename all other account names.

If we remove the dash from the organization name, we have to rename the npm package names as well (`@inexor-game/...` --> `@inexorgame/...`).

From product perspective I would like to force consistent naming. We are offering two products (`Inexor Core` and `Inexor Flex`) and `Inexor` is an essential and homogeneous part of it. Only both words together are forming the brand. I know it's repetitive but 

> [...] in my opinion it would be another inconsistency if two, but only two, repositories do have the "inexor-" prefix

True, then we should rename `data` and `data-additional`to `inexor-repo-core` and `inexor-repo-additional` or `inexor-media-core` and `inexor-media-additional`. In my opinion the name `data` is a leftover from the good old sauer days where we hadn't media repositories. The `data` folder of sauer contained everything that wasn't C/C++ code. This won't be the case anymore: the media repositories shall only contain media files (textures, models, maps, ...) and no user interfaces, cubescript and configuration files.

**Comment by a-teammate** [comment]
Reasonable compromise​.

Maybe the inexorgame name appeared bc inexor-game contained a forbidden character for a name on some platforms.

I agree with the media stuff.

We shouldn't forgot to rename all occurrences in the wiki as well.

So the required steps are:
- [ ] clone all repos. Replace all occurrences
- [ ] clone the wiki. Replace all occurrences
- [ ] search the issue tracker for the old names (??)
- [ ] move all miss-named accounts
* take care when merging feature branches

**Comment by Croydon** [comment]
> True, then we should rename data and data-additional to inexor-media-essential and inexor-media-additional

Please don't. It's terrible to have inexor in every repo name repeated, there is a reason why we have an organisation in the first place. There are also reasons why git clone has a parameter where you can specify a local name. We can update the Wiki respectively, by simply adding an inexor-core behind the clone command if that is getting rid of your concerns, that the default "core" or "code" is too generic @Hanack. Also in the end of the day it should be completely up to every person how to organise files.

Just imagine every GitHub organisation would do that.

**Comment by aschaeffer** [comment]
> [...] in my opinion it would be another inconsistency if two, but only two, repositories do have the "inexor-" prefix. [...]

You complained about inconsistencies so I tried to make a suggestion which is consistent by prefixing everything with `inexor`followed by the type of the repository like `ui` or `media` or whatever then followed by the actual name of the ui or media or whatever. So that there is a fixed naming scheme `inexor-{type}-{name}` for every repository.

For me the `inexor-` prefix is not that important for media and ui repositories. But we should at least stick with `{type}-{name}` for every `ui` or `media` or whatever repository.

Actually my main intention is that the `Inexor Core` and  `Inexor Flex` repositories are fully named (`inexor-core` and `inexor-flex`). The reason is that we really should make them brands:

> [...] We are offering two products (`Inexor Core` and `Inexor Flex`) and Inexor is an essential and homogeneous part of it. Only both words together are forming the brand. I know it's repetitive.

**Comment by Croydon** [comment]
> You complained about inconsistencies 

Yes, I pointed that out in your suggestion, but as previously explained I am not a fan of the inexor- prefix alltogether.

> The reason is that we really should make them brands:

But shouldn't Flex be invisible for the average player? So from my understanding Flex is a brand for only developers and server admins. (Yes, brands can exists for these target groups, just pointing out again). And for the average player we have only one brand: Inexor. They don't care for internals.

Anyway, if we don't find further consensus, I'm willing to compromise and say let's go for:
 * inexor-game -> inexorgame
 * code -> inexor-core 
 * flex -> inexor-flex
 * data -> media-core
 * data-additional -> media-additional
 * data-playground -> media-playground
 * data-src -> media-src
 * InexorGlueGen -> gluegen
 * leave the rest as it is

I like the {type}-{name} schema. I also used it also for the ci repositories.

How does that sound to everyone?

(And I will roll my eyes everytime I have to type the URL to our core repo 😁)

**Comment by Fohlen** [comment]
I did like the inexor-game prefix very much
Why would flex become inexor-flex, and code become inexor-core instead inexor-game/flex and inexor-game/core

**Comment by Croydon** [comment]
> I did like the inexor-game prefix very much

When we said prefix, we mean the suggestion "inexor-" prefix in the repository name, the organisation name stays of course (or the minus is getting removed from it, but stays then).

**Comment by Croydon** [comment]
I have to correct my past-self: After reading redirections again I guess it doesn't matter if we first rename repos and then organisation or the other way around.

Since renaming the org seems to be the only agreement for sure so far I would start now with that.

**Comment by Croydon** [comment]
> search the issue tracker for the old names (??)

This can be done only manually and has very little use. I'm strongly against doing that.

**Comment by a-teammate** [comment]
Search for "inexor-game in:comments"

**Comment by Croydon** [comment]
okay, but searching without editing is pointless and editing only works manually? or someone needs to write a script for that..

**Comment by a-teammate** [comment]
@Croydon actually im in favor of Hanacks proposal.
But if we really do your 

>  * inexor-game -> inexorgame
>  * code -> inexor-core 
>  * flex -> inexor-flex
>  * data -> media-core
>  * data-additional -> media-additional
>  * data-playground -> media-playground
>  * data-src -> media-src
>  * InexorGlueGen -> gluegen
>  * leave the rest as it is

im strongly against two points:
* InexorGlueGen -> gluegen
and
* data -> media-core
InexorGlueGen should be called like that, its a distinct application.
`media-core` is a weak name since it steals `core` from `InexorCore`.
`media-essential` is better.

Furthermore `media-src` is imo not meaningful enough. `media-sources` skips the step in your brain to unfold "src" to sth meaningful.

**However** i would really like if just everything which is really for the game is named like that.
if "inexorgame" would have been called "inexor" this would not be a problem, but since the name "inexor" is taken on GitHub we have to improvise.
And to depend on the discipline of everybody to keep a good ordering on their local harddrive when cloning is not really sth anyone should ever do ;D

**Hence** i really prefer the verbosity of Hanacks suggestion, although it needs some adaption to read (it does not really look cool, but it adds clarity)


> * https://github.com/inexorgame/inexor-core
> * https://github.com/inexorgame/inexor-flex
> * https://github.com/inexorgame/inexor-media-essential (or inexor-media-main)
> * https://github.com/inexorgame/inexor-media-additional
> * https://github.com/inexorgame/inexor-ui-flex
> * https://github.com/inexorgame/inexor-ui-core-console
> * https://github.com/inexorgame/inexor-ui-core-hud
> * https://github.com/inexorgame/inexor-ui-core-menu
> * https://github.com/inexorgame/inexor-ui-core-interface

with the addition of 
* https://github.com/inexorgame/inexor-media-experimental
* https://github.com/inexorgame/inexor-media-sources

and leaving the rest as is.
ofc `inexor-core` and `inexor-flex` are the most important ones here.
I guess we need some more voices to break the tie or we just skip "inexor-" for anything but the both above..

**Comment by Croydon** [comment]
> im strongly against two points:
    InexorGlueGen -> gluegen

Okay, but the whole point is about consistency. InexorGlueGen is our only repository with upper cases (conan packages with external names excluded) AND the only compount name without hypens between each part.

New suggestion: What about inexor-gluegen?

> Furthermore media-src is imo not meaningful enough. media-sources skips the step in your brain to unfold "src" to sth meaningful.

Agreed.

> media-core is a weak name since it steals core from InexorCore.
media-essential is better.

We called the main media repo already media-core long before we started calling our main code repo core. And, again: consistency. Why one time main and one time core?

**Comment by a-teammate** [comment]
InexorGlueGen is sth really distinct.
It could be named anything.
koraa once even suggested the name `dove` for it, since its actually just a dependency we have. A totally different project if you will.

media-core:
core should disappear to make the name "core" (if you omit the "inexor" when you abbreviate) unambiguous.

**Comment by Croydon** [comment]
> media-core:
core should disappear to make the name "core" (if you omit the "inexor" when you abbreviate) unambiguous.

Alright. But then I suggest to just name it "media", it breaks for now with the {media}-{type} schema, but makes more sense in the long run. media(-core) will be the only media repository from us at some point.  
We just have one website, so we don't call it site-core as well.

**Comment by a-teammate** [comment]
why not rename it after it gets the only media repo we have?

**Comment by Croydon** [comment]
> why not rename it after it gets the only media repo we have?

Because we break URLs again then, and only one redirect works, so the current URLs would break for good with a second rename. Also renaming shouldn't become a regular thing.

**Comment by a-teammate** [comment]
okay, seems legit.

so to summarize what we already agreed on:
* https://github.com/inexorgame/inexor-core
* https://github.com/inexorgame/inexor-flex

what we did not yet agree on:

the `inexor-` prefix for the following repos:

* https://github.com/inexorgame/inexor-media
* https://github.com/inexorgame/inexor-media-additional
* https://github.com/inexorgame/inexor-media-experimental
* https://github.com/inexorgame/inexor-media-sources
* https://github.com/inexorgame/inexor-ui-flex
* https://github.com/inexorgame/inexor-ui-core-console
* https://github.com/inexorgame/inexor-ui-core-hud
* https://github.com/inexorgame/inexor-ui-core-menu
* https://github.com/inexorgame/inexor-ui-core-interface

I think we should agree asap since some stuff is broken until we do. (e.g. we did not yet replace the links in the code repo)

**Comment by aschaeffer** [comment]
Actually I think the probability of providing multiple media repositories is higher than having only one. It could be that we break down things in the structure of media repositories as well. For example to create a media repository for a game mode or a group of game modes.

**Comment by a-teammate** [comment]
okay.
People srsly lets make this decision.
two open points:
*  `media-essential` vs `media` for the main data repo
   * react with :tada: for `media-essential` and with :heart: for `media`
*  `inexor-` prefix for the `ui` and `media` repos: yes or no.
   * react with :+1: for the `inexor-` prefix and with :-1: for no

I'd suggest the deadline to end Sunday 25.06.2017 for this voting. Anyone can vote.

**Comment by Croydon** [comment]
If not just media then please media-main instead of media-essential. Still shorter.

**Comment by Fohlen** [comment]
`media` is fine, and `media-essential` is even finer.
A comprehensive name is very important to quickly identify things.
`main` does not tell you anything about the repo's function, `main` of what? `main` of submodules? `main` of configs? `essential` is much better.

**Comment by Croydon** [comment]
> main does not tell you anything about the repo's function, main of what? main of submodules? main of configs?

main of media. of course, that is why it would be named media-main and not just main. You can ask the very same question with essential.

essential does not tell you anything about the repo's function, essential of what? essential of submodules? essential of configs?

No, essential media stuff, of course.

**Comment by Fohlen** [comment]
> main of media. of course, that is why it would be named media-main and not just main. You can ask the very same question with essential.

not "of course" here.

> essential does not tell you anything about the repo's function, essential of what? essential of submodules? essential of configs?

I disagree. Media essential tells you: you want to get started with inexor? These media files are essential to the game.

**Comment by Croydon** [comment]
Another note and probably my last one: We once renamed the "src" directory in our core repository to "inexor" so that it couldn't collide with other source directories (why the hell should anyone want to mix it up with other stuff anyway? keep your source clean!).
Now we have a inexorgame/inexor-core/inexor directory. Oh, and a "src" directory in flex. And we want to rename "media-src" to "media-source" because "src" needs another "translation" step in the brain.
...

My point is, I can't see the overall goal of how you want to name things. Consistency can be achieved in a lot of ways, it doesn't explain the naming schemes, which seem to make names longer without making them easier in any way. I would even argue harder, because they are longer and have redundant information. How to name stuff locally and how and where to save stuff is up to each and every one individually.

**Comment by a-teammate** [comment]
> We once renamed the "src" directory in our core repository to "inexor" so that it couldn't collide with other source directories (why the hell should anyone want to mix it up with other stuff anyway? keep your source clean!).

no, it should not collide with system headers ("net/", but i srsly doubt this step was necessary, but that had been a totally different discussion. id be open for changing it back, but plz another issue)

> My point is, I can't see the overall goal of how you want to name things.

Well this whole issue is about making compromises.. but yes, "src" could be "source" everywhere.

**Comment by a-teammate** [comment]
> okay.
> People srsly lets make this decision.
> two open points:
> 
> media-essential vs media for the main data repo
> react with :tada: for media-essential and with :heart: for media
> inexor- prefix for the ui and media repos: yes or no.
> react with :+1: for the inexor- prefix and with :-1: for no
> I'd suggest the deadline to end Sunday 25.06.2017 for this voting. Anyone can vote
> 

okay, its: 2  :+1: , 2  :-1:, 4 :tada: and 2 :heart: 
this means:

* the main media repo will be named `media-essential`
* patt for the `inexor-`-prefix
   * aschaeffer just said it does not bother him much
      -> no `inexor`prefix for media and ui repos

Go!
